gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +967,920 @@ public void testListDnsRecords() {
+      }
+      // field options
+      changes = ImmutableList.copyOf(resultAdditions.get().iterateAll());
+      change = changes.get(1);
+      assertEquals(CHANGE_ADD_ZONE1.additions(), change.additions());
+      assertTrue(change.deletions().isEmpty());
+      assertEquals("1", change.generatedId());
+      assertNull(change.startTimeMillis());
+      assertNull(change.status());
+      changes = ImmutableList.copyOf(resultDeletions.get().iterateAll());
+      change = changes.get(2);
+      assertTrue(change.additions().isEmpty());
+      assertNotNull(change.deletions());
+      assertEquals("2", change.generatedId());
+      assertNull(change.startTimeMillis());
+      assertNull(change.status());
+      changes = ImmutableList.copyOf(resultId.get().iterateAll());
+      change = changes.get(1);
+      assertTrue(change.additions().isEmpty());
+      assertTrue(change.deletions().isEmpty());
+      assertEquals("1", change.generatedId());
+      assertNull(change.startTimeMillis());
+      assertNull(change.status());
+      changes = ImmutableList.copyOf(resultTime.get().iterateAll());
+      change = changes.get(1);
+      assertTrue(change.additions().isEmpty());
+      assertTrue(change.deletions().isEmpty());
+      assertEquals("1", change.generatedId());
+      assertNotNull(change.startTimeMillis());
+      assertNull(change.status());
+      changes = ImmutableList.copyOf(resultStatus.get().iterateAll());
+      change = changes.get(1);
+      assertTrue(change.additions().isEmpty());
+      assertTrue(change.deletions().isEmpty());
+      assertEquals("1", change.generatedId());
+      assertNull(change.startTimeMillis());
+      assertEquals(ChangeRequest.Status.DONE, change.status());
+    } finally {
+      clear();
+    }
+  }
+
+  @Test
+  public void testListDnsRecordSetsBatch() {
+    try {
+      Zone zone = DNS.create(ZONE1);
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<Page<RecordSet>> result = batch.listRecordSets(zone.name());
+      batch.submit();
+      ImmutableList<RecordSet> recordSets = ImmutableList.copyOf(result.get().iterateAll());
+      assertEquals(2, recordSets.size());
+      ImmutableList<RecordSet.Type> defaultRecords =
+          ImmutableList.of(RecordSet.Type.NS, RecordSet.Type.SOA);
+      for (RecordSet recordSet : recordSets) {
+        assertTrue(defaultRecords.contains(recordSet.type()));
+      }
+      // field options
+      batch = DNS.batch();
+      DnsBatchResult<Page<RecordSet>> ttlResult = batch.listRecordSets(zone.name(),
+          Dns.RecordSetListOption.fields(Dns.RecordSetField.TTL));
+      DnsBatchResult<Page<RecordSet>> nameResult = batch.listRecordSets(zone.name(),
+          Dns.RecordSetListOption.fields(Dns.RecordSetField.NAME));
+      DnsBatchResult<Page<RecordSet>> recordsResult = batch.listRecordSets(zone.name(),
+          Dns.RecordSetListOption.fields(Dns.RecordSetField.DNS_RECORDS));
```

Comment:
Should we rename this as per an earlier discussion to rename DnsRecord to RecordSet?